upp/classes/components.py

```diff
@@ -63,7 +67,12 @@ def check_num_jets(
         self, num_req, sampling_frac=None, cuts=None, silent=False, raise_error=True
     ):
         # Check if num_jets jets are aviailable after the cuts and sampling fraction
-        total = self.reader.estimate_available_jets(cuts, self.num_jets_estimate)
+        if self.num_jets_estimate_num <= 0:
+            self.setup_reader(**self.reader_replicate_kwargs)
```

Comment:
why are you creating new readers here? the results won't be the same with successive calls but the behaviour is still deterministic. 

If you want to have the same number returned each time, I'd suggest to change the logic in the reader (moving the [rng def](https://github.com/umami-hep/atlas-ftag-tools/blob/main/ftag/hdf5/h5reader.py#L30)) , so that we don't have to create a new one every time we want to use it.

but it feels like to get the tests passing we should only require deterministic results, not necessarily that the same numbers are returned for successive calls, so I'm wondering why you add this restriction?

Reply:
As the reader returns not the same at each of the consecutive calls one gets different results for `estimate_available_jets()` in different parts of the code and it leads to bugs where one uses one number to estimate the correct resampling fraction and another value to check if there are sufficient jets if we use this resampling fraction. This was the main reason why the tests were not passing. 
This can be fixed in three different ways:
1. only calling `estimate_available_jets` once and saving it result oping that nobody call reader again in-between two steps (bad solution)
2. Reinitialise the reader resetting its seed (does not require to mangle with atlas-ftag-tools code and update the dependencies everywhere)
3. Adding rng seed setting in `estimate_available_jets` itself to make it invariant on the call number. Will require updating atlas-ftag-tools everywhere.

I opted for solution 2 as the simplest. Creating a new reader is a fast process requires barely any resources, so it is not a performance issue. I solve it in a different way (e.g. 3) if you prefer.

Reply:
I prefer to fix this in atlas ftag tools. The workaround here isn't so bad, but don't doesn't generalise others who may use the tools package.

Do you think something like this would suffice?
https://github.com/umami-hep/atlas-ftag-tools/pull/67

Reply:
Yes, resetting rng in `estimate_available_jets` should perfectly solve the issue. I will make all necessary changes in UPP if you do that merge and release a new version of ftag-tools.

Reply:
Great, thanks! I have published a new release https://github.com/umami-hep/atlas-ftag-tools/releases/tag/v0.1.15

Reply:
I have now updated it to work with version 1.16.1 of ftag tools (removed the reinitialization of reader).
Is it fine to merge now?

Reply:
Hi, I updated it to version 1.16.1 in ftag-tools. I removed the reinitialization of the reader

Reply:
Is it ok to merge this now?